[source-git update] git cherry-pick fails due to staged changes

```
$ cd rpms; git clone https://git.centos.org/rpms/apr.git
$ cd ../source-git; git clone ssh://git@git.stg.centos.org/source-git/apr.git
$ cd ..; CONTAINER_CMD="dist2src -v convert rpms/apr:c8s source-git/apr:c8" make run
docker run \
	-ti --rm \
	-v /home/jpopelka/git/packit/dist-git-to-source-git/dist2src:/usr/local/lib/python3.6/site-packages/dist2src:Z \
	-v /home/jpopelka/git/packit/dist-git-to-source-git/packitpatch:/usr/bin/packitpatch:Z \
	-v /home/jpopelka/git/packit/dist-git-to-source-git/macros.packit:/usr/lib/rpm/macros.d/macros.packit:Z \
	-v /home/jpopelka/git/packit/dist-git-to-source-git/tests:/tests:Z \
	-v /home/jpopelka/git/packit/dist-git-to-source-git/rpms:/workdir/rpms:Z \
	-v /home/jpopelka/git/packit/dist-git-to-source-git/source-git:/workdir/source-git:Z \
	-u 1000 \
	 \
	dist2src dist2src -v convert rpms/apr:c8s source-git/apr:c8
convert(origin='rpms/apr:c8s', dest='source-git/apr:c8')
Running command get_sources.sh in /workdir/rpms/apr
rpmbuild stderr = + umask 022
+ cd /workdir/rpms/apr/BUILD
+ cd /workdir/rpms/apr/BUILD
+ rm -rf apr-1.6.3
+ /usr/bin/bzip2 -dc /workdir/rpms/apr/SOURCES/apr-1.6.3.tar.bz2
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd apr-1.6.3
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ /usr/bin/git init
+ /usr/bin/git add -f .
+ /usr/bin/git commit -q --allow-empty -a -m 'apr-1.6.3 base'
+ echo 'Patch #1 (apr-1.6.3-r1834495.patch):'
+ /usr/bin/packitpatch '%{1}' '%{2}' --fuzz=0 --no-backup-if-mismatch --no-backup-if-mismatch -p1 -b --suffix .r1834495 --fuzz=0
+ echo 'Patch #2 (apr-1.2.2-locktimeout.patch):'
+ /usr/bin/packitpatch '%{1}' '%{2}' --fuzz=0 --no-backup-if-mismatch --no-backup-if-mismatch -p1 -b --suffix .locktimeout --fuzz=0
+ echo 'Patch #3 (apr-1.2.2-libdir.patch):'
+ /usr/bin/packitpatch '%{1}' '%{2}' --fuzz=0 --no-backup-if-mismatch --no-backup-if-mismatch -p1 -b --suffix .libdir --fuzz=0
+ echo 'Patch #4 (apr-1.2.7-pkgconf.patch):'
+ /usr/bin/packitpatch '%{1}' '%{2}' --fuzz=0 --no-backup-if-mismatch --no-backup-if-mismatch -p1 -b --suffix .pkgconf --fuzz=0
+ echo 'Patch #5 (apr-rh1539844.patch):'
+ /usr/bin/packitpatch '%{1}' '%{2}' --fuzz=0 --no-backup-if-mismatch --no-backup-if-mismatch -p1 -b --suffix .rh1539844 --fuzz=0
+ echo 'Patch #6 (apr-1.4.8-deepbind.patch):'
+ /usr/bin/packitpatch '%{1}' '%{2}' --fuzz=0 --no-backup-if-mismatch --no-backup-if-mismatch -p1 -b --suffix .deepbind --fuzz=0
+ exit 0

The repo is not dirty, nothing to commit.
Fetch the dist-git %prep branch to source-git branch updates.
Traceback (most recent call last):
  File "/usr/local/bin/dist2src", line 33, in <module>
    sys.exit(load_entry_point('dist-git-to-source-git==0.1.dev218+g8cf46c8.d20201019', 'console_scripts', 'dist2src')())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/dist2src/cli.py", line 83, in wrapper
    ret = func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/dist2src/cli.py", line 226, in convert
    d2s.convert(origin_branch, dest_branch)
  File "/usr/local/lib/python3.6/site-packages/dist2src/core.py", line 391, in convert
    from_branch=TEMP_SG_BRANCH, to_branch=dest_branch, theirs=update
  File "/usr/local/lib/python3.6/site-packages/dist2src/core.py", line 140, in cherry_pick_base
    self.repo.git.cherry_pick(f"{from_branch}~{num_commits - 1}", **git_options)
  File "/usr/local/lib/python3.6/site-packages/git/cmd.py", line 542, in <lambda>
    return lambda *args, **kwargs: self._call_process(name, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/git/cmd.py", line 1005, in _call_process
    return self.execute(call, **exec_kwargs)
  File "/usr/local/lib/python3.6/site-packages/git/cmd.py", line 822, in execute
    raise GitCommandError(command, status, stderr_value, stdout_value)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git cherry-pick updates~6
  stderr: 'error: your local changes would be overwritten by cherry-pick.
hint: commit your changes or stash them to proceed.
fatal: cherry-pick failed'
make: *** [Makefile:41: run] Error 1
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,6 +1,7 @@
 # Copyright Contributors to the Packit project.
 # SPDX-License-Identifier: MIT
 import os
+import subprocess
 from pathlib import Path
 
 from click.testing import CliRunner
@@ -86,3 +87,25 @@ def run_packit(*args, working_dir=None, **kwargs):
     with cwd(working_dir):
         cli_runner = CliRunner()
         cli_runner.invoke(packit_base, *args, catch_exceptions=False, **kwargs)
+
+
+def clone_package(
+    package_name: str,
+    dist_git_path: str,
+    branch: str = "c8s",
+    namespace: str = "rpms",
+    stg: bool = False,
+):
+    """
+    clone selected package from git.[stg.]centos.org
+    """
+    subprocess.check_call(
+        [
+            "git",
+            "clone",
+            "-b",
+            branch,
+            f"https://git{'.stg' if stg else ''}.centos.org/{namespace}/{package_name}.git",
+            dist_git_path,
+        ]
+    )

--- a/tests/test_convert.py
+++ b/tests/test_convert.py
@@ -12,20 +12,12 @@ from tests.conftest import (
     TEST_PROJECTS_WITH_BRANCHES_SINGLE_COMMIT,
     run_dist2src,
     run_packit,
+    clone_package,
 )
 
 
 def convert_repo(package_name, dist_git_path, sg_path, branch="c8s"):
-    subprocess.check_call(
-        [
-            "git",
-            "clone",
-            "-b",
-            branch,
-            f"https://git.centos.org/rpms/{package_name}.git",
-            dist_git_path,
-        ]
-    )
+    clone_package(package_name, str(dist_git_path), branch=branch)
     run_dist2src(
         ["-vvv", "convert", f"{dist_git_path}:{branch}", f"{sg_path}:{branch}"]
     )

--- a/tests/test_dist2src.py
+++ b/tests/test_dist2src.py
@@ -7,6 +7,7 @@ from pathlib import Path
 import pytest
 
 from dist2src.core import Dist2Src
+from tests.conftest import clone_package
 from tests.test_convert import run_dist2src
 
 this_dir = Path(__file__).parent
@@ -144,10 +145,7 @@ def test_no_backup(tmp_path: Path):
     d = tmp_path / "d"
     d.mkdir()
     dist_git_path = d / package_name
-    subprocess.check_call(
-        ["git", "clone", f"https://git.centos.org/rpms/{package_name}.git"], cwd=d
-    )
-    subprocess.check_call(["git", "checkout", "c8"], cwd=dist_git_path)
+    clone_package(package_name, str(dist_git_path), branch="c8")
     b = dist_git_path / "BUILD" / "hyperv-daemons-0"
     run_dist2src(["-v", "run-prep", str(dist_git_path)], working_dir=dist_git_path)
     assert b.is_dir()

--- a/tests/test_update.py
+++ b/tests/test_update.py
@@ -180,3 +180,40 @@ def test_update_source(tmp_path: Path, package_name, branch, old_version):
                 srpm_path,
             ]
         )
+
+
+def test_update_apr(tmp_path: Path):
+    """
+    Jirka found an issue that we cannot update several packages
+    which were created by an old version of d2s, one of them is apr
+    """
+    package_name = "apr"
+    dist_git_path = tmp_path / "d" / package_name
+    dg_branch = "c8s"
+    source_git_path = tmp_path / "s" / package_name
+    sg_branch = "c8"
+    clone_package(package_name, str(dist_git_path), branch=dg_branch)
+    clone_package(
+        package_name,
+        str(source_git_path),
+        branch=sg_branch,
+        namespace="source-git",
+        stg=True,
+    )
+    run_dist2src(
+        [
+            "-vvv",
+            "convert",
+            f"{dist_git_path}:{dg_branch}",
+            f"{source_git_path}:{sg_branch}",
+        ]
+    )
+    run_packit(
+        [
+            "--debug",
+            "srpm",
+        ],
+        working_dir=source_git_path,
+    )
+    srpm_path = next(source_git_path.glob("*.src.rpm"))
+    assert srpm_path.exists()

--- a/tests/test_update.py
+++ b/tests/test_update.py
@@ -13,6 +13,7 @@ from tests.conftest import (
     TEST_PROJECTS_WITH_BRANCHES_SINGLE_COMMIT,
     run_dist2src,
     run_packit,
+    clone_package,
 )
 
 
@@ -27,16 +28,7 @@ def test_update(tmp_path: Path, package_name, branch):
     dist_git_path.mkdir(parents=True)
     sg_path.mkdir(parents=True)
 
-    subprocess.check_call(
-        [
-            "git",
-            "clone",
-            "-b",
-            branch,
-            f"https://git.centos.org/rpms/{package_name}.git",
-            dist_git_path,
-        ]
-    )
+    clone_package(package_name, str(dist_git_path), branch=branch)
     subprocess.check_call(
         ["git", "reset", "--hard", "HEAD~1"],
         cwd=dist_git_path,
@@ -88,16 +80,7 @@ def test_update_from_same_commit(tmp_path: Path, package_name, branch):
     dist_git_path.mkdir(parents=True)
     sg_path.mkdir(parents=True)
 
-    subprocess.check_call(
-        [
-            "git",
-            "clone",
-            "-b",
-            branch,
-            f"https://git.centos.org/rpms/{package_name}.git",
-            dist_git_path,
-        ]
-    )
+    clone_package(package_name, str(dist_git_path), branch=branch)
 
     run_dist2src(
         ["-vvv", "convert", f"{dist_git_path}:{branch}", f"{sg_path}:{branch}"]
@@ -157,16 +140,7 @@ def test_update_source(tmp_path: Path, package_name, branch, old_version):
     dist_git_path.mkdir(parents=True)
     sg_path.mkdir(parents=True)
 
-    subprocess.check_call(
-        [
-            "git",
-            "clone",
-            "-b",
-            branch,
-            f"https://git.centos.org/rpms/{package_name}.git",
-            dist_git_path,
-        ]
-    )
+    clone_package(package_name, str(dist_git_path), branch=branch)
     subprocess.check_call(
         ["git", "reset", "--hard", old_version],
         cwd=dist_git_path,

